CMakeLists.txt

```diff
@@ -74,13 +74,16 @@ if (USE_RUNPATH)
    endif()
 endif (USE_RUNPATH)
 
-
-if (NOT DEFINED CMAKE_CXX_FLAGS)
-  SET( CMAKE_CXX_FLAGS "-pipe -std=c++0x -pedantic -Wall -Wextra -Wformat-nonliteral -Wcast-align -Wpointer-arith -Wmissing-declarations -Wcast-qual -Wshadow -Wwrite-strings -Wchar-subscripts -Wredundant-decls")
-endif()
-
-SET( CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_DEBUG}   -O0 -DDEBUG  -ggdb3")
-SET( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_RELEASE} -O3 -DNDEBUG -mtune=native")
+if (MSVC)
+  add_definitions( "/W3 /D_CRT_SECURE_NO_WARNINGS /wd4996 /wd4244 /wd4267")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
```

Comment:
My CMake skills are quite rudimentary; but as I understand this code will not set `_DEBUG` and `_RELEASE` flags. Wouldn't this be more in line with the gnu solution:
```
if (MSVC)
    add_definitions( "/W3 /D_CRT_SECURE_NO_WARNINGS /wd4996 /wd4244 /wd4267")
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")

    set(CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_DEBUG} -extra -windows -debug -flags)
    set(CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_RELEASE} -extra -windows -release -flags) 
endif()
```